README.md

```diff
@@ -26,18 +40,140 @@ A slackbot which reads an excel file from sharepoint and returns a specific colu
 ```
 
 ### Start
-1. Go to your project folder and type npm start
-2. CTRL + C to stop the bot 
+1. Make sure you have downloaded all the needed dependencies 
+2. Make sure you're in the project root
+```yaml
+cd /<your-folder-name>/slackbot-sharepoint-excel-reader
+```
+3. type npm start, it should create a folder "build" and a file inside of it called "app.js" and give you this output in your terminal
+```yaml
+npm start
+
+> slackbot0.0.1 start
+> tsc && node build/app.js
+
+⚡️ Bolt app is running in Socket Mode!
+```
+
+Go into slack and try typing /"your-command" in any channel. The bot should respond "Hello + your name 👋"
+You can stop the bot by pressing CTRL + C in your terminal. After that, the bot should stop responding to your command
+
+this is what your project structure should look like:
+```yaml
+Slackbot-Sharepoint-Excel-Reader
+│
+├── build               
+│   └── app.js
+│
+├── node_modules        
+│
+├── src                  
+│   └── app.ts
+│
+├── .env                 
+├── .gitignore          
+├── package-lock.json    
+├── package.json         
+├── README.md          
+└── tsconfig.json
+```
+## Hosting on AWS Lambda and deploying the bot with serverless
+### Installing and setting up AWS and serverless
+1. [Sing in into AWS console](https://aws.amazon.com/console/)
+2. Click on your name on the top right corner and click security credentials
+3. Create an access key and take write them down / download them
+4. install AWS CLI [see documentation here](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
+5. Configure your AWS CLI tool to use your access keys
+```yaml
+aws configure
+
+AWS Access Key ID [None]: <copy & paste your access key> 
+AWS Secret Access Key [None]: <copy & paste your secret key> 
+Default region name [None]: <choose whatever you want>
+Default output format [None]: JSON 
+```
+6. install serverless and serverless offline for local testing
+```yaml
+npm i serverless -g
+npm install serverless-offline --save-dev
+```
+### Hosting
+> At this point you should have files like serverless.yml and handler.js in src
+```yaml
+Slackbot-Sharepoint-Excel-Reader
+│
+├── .serverless/              
+│
+├── build/                     
+│   ├── app.js              
+│   └── handler.js          
+│
+├── node_modules/             
+│
+├── src/                       
+│   ├── app.ts               
+│   └── handler.ts              
+│
+├── .env                     
+├── .gitignore                 
+├── package-lock.json        
+├── package.json              
+├── README.md                
+├── serverless.yml           
+└── tsconfig.json        
+```
+1. Go to Slack api -> socket mode and disable it
+2. type serverless deploy 
+3. When its ready, it will give you an endpoint looking something like "https://6psdswxvx2.execute-api.eu-north-1.amazonaws.com/dev/slack/events"
+> Remember that this will change everytime you do serverless deploy so you have to do the next part again each time.
```

Comment:
This should not change if you do serverless deploy. It only changes if you remove the existing cloudformation stack and then redeploy. Redeploying on top of existing CF stack should retain all names and addresses